Fixed the missing checkbox labels

shivansh-bhatnagar18 opened the pull request:
## Summary
### Description of the change(s) you made
I have fixed the missing checkout labels in the content library that were used for filtering out the content for the users to prevent confusion and possible unintended selections. Fixes the issue #4525 

### Manual verification steps performed
1. Step 1 ) Checked the working of the component by running the app locally

### Screenshots (if applicable)
[Screencast from 2024-04-20 13-49-01.webm](https://github.com/learningequality/studio/assets/117429248/b141351c-ea32-49ab-85af-336fae499be9)

PR process:

- [ ] If this is an important user-facing change, PR or related issue the `CHANGELOG` label been added to this PR. Note: items with this label will be added to the [CHANGELOG](https://github.com/learningequality/studio/blob/master/CHANGELOG.md) at a later time
- [ ] If this includes an internal dependency change, a link to the diff is provided
- [ ] The `docs` label has been added if this introduces a change that needs to be updated in the [user docs](https://kolibri-studio.readthedocs.io/en/latest/index.html)?
- [ ] If any Python requirements have changed, the updated `requirements.txt` files also included in this PR
- [ ] Opportunities for using Google Analytics here are noted
- [ ] Migrations are [safe for a large db](https://www.braintreepayments.com/blog/safe-operations-for-high-volume-postgresql/)

Studio-specifc:

- [ ] All user-facing strings are translated properly
- [ ] The `notranslate` class been added to elements that shouldn't be translated by Google Chrome's automatic translation feature (e.g. icons, user-generated text)
- [ ] All UI components are LTR and RTL compliant
- [ ] Views are organized into `pages`, `components`, and `layouts` directories [as described in the docs](https://github.com/learningequality/studio/blob/vue-refactor/docs/architecture.md#where-does-the-frontend-code-live)
- [ ] Users' storage used is recalculated properly on any changes to main tree files
- [ ] If there new ways this uses user data that needs to be factored into our [Privacy Policy](https://github.com/learningequality/studio/tree/master/contentcuration/contentcuration/templates/policies/text), it has been noted.


Testing:

- [ ] Code is clean and well-commented
- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Any new interactions have been added to the [QA Sheet](https://docs.google.com/spreadsheets/d/1HF4Gy6rb_BLbZoNkZEWZonKFBqPyVEiQq4Ve6XgIYmQ/edit#gid=0)
- [ ] Critical and brittle code paths are covered by unit tests
___

### Reviewer's Checklist
#### This section is for reviewers to fill out.

- [ ] Automated test coverage is satisfactory
- [ ] PR is fully functional
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependency files were updated if necessary (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Contributor is in AUTHORS.md

bjester reviewed (changes requested):
Thank you @shivansh-bhatnagar18 for your interest. Before opening PRs, please ask for assignment first, so that we can help guide you.

Unfortunately, this isn't the correct fix.

MisRob commented:
For context, it may be another regression from https://github.com/learningequality/studio/pull/4472, which we already received some fixes for, such as https://github.com/learningequality/studio/pull/4512

shivansh-bhatnagar18 commented:
@bjester Actually for the multiselect component (which controls Licenses and Formats) which was easier to replace with the KCheckbox, I replaced the component with the Kolibri Design Systems but for complex components like Languages dropdown which requires Autocomplete and Search Functionalities for which component has been made in Kolibri design systems, I have fixed the missing labels issue.

bjester commented on contentcuration/contentcuration/frontend/shared/views/form/MultiSelect.vue:
```diff
@@ -20,14 +20,14 @@
+            :label="getText(item)"
+            :value="item"
+            :style="getEllipsisStyle()"
+            :ripple="false"
```
There isn't a `ripple` prop for `KCheckbox`. Also, it wouldn't seem this is preserving the `notranslate` class?

bjester commented:
Hi @shivansh-bhatnagar18,

Seems there's a remaining linting issue on contentcuration/contentcuration/frontend/channelList/views/Channel/components/LanguageFilter.vue

shivansh-bhatnagar18 commented:
Hi @bjester, I fixed the linting issues. There are certain things, 
1) The KCheckbox is previously used in kolibri studio and I see usage of ripple prop for the same. (Attached below) Although I can't find it mentioned in the documentation of Kolibri Design Systems
![Screenshot from 2024-05-21 01-22-53](https://github.com/learningequality/studio/assets/117429248/22969bed-6203-4479-8cb1-92695449a316)
2) Now the Kcheckbox also preserves the `notranslate` class so that it automatically prevents translation

MisRob commented:
Hi @shivansh-bhatnagar18, thanks for following up. Regarding the `ripple` prop usage in another places, that's most likely dead code. There's nothing like that in `KCheckbox` https://github.com/learningequality/kolibri-design-system/blob/release-v4/lib/KCheckbox.vue so it'd be best to remove it.

MisRob commented:
Hi @shivansh-bhatnagar18, just chiming in to ask how things are going, is there something we can support with?

rtibbles commented:
Closing in favour of the more recently updated https://github.com/learningequality/studio/pull/4629
